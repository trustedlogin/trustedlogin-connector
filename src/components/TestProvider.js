import { SettingsProvider } from "../hooks/useSettings";
import ViewProvider from "../hooks/useView";
export const testTeam = {
  account_id: "1",
  private_key: "asda",
  public_key: "fsaffff",
  helpdesk: "helpscout",
  approved_roles: [],
  helpdesk_settings: {
    helpscout: {
      secret: "9594aef90fce5450",
      callback:
        "https://trustedlogin.ngrok.io/?trustedlogin=1&action=trustedlogin_webhook&provider=helpscout&ak_account_id=asdsad&_tl_ak_nonce=f8bb651568",
    },
  },
  id: 0,
};
const mockApi = {
  getSettings: async () => {
    return {
      teams: [testTeam],
    };
  },
  updateSettings: async () => {},
};

/**
 * All providers in one.
 *
 * ONLY use for testing.
 */
const TestProvider = ({
  children,
  api = mockApi,
  hasOnboarded = true,
  initialTeams = [testTeam],
  initialIntegrationSettings = null,
}) => (
  <SettingsProvider
    api={api}
    hasOnboarded={hasOnboarded}
    initialTeams={initialTeams}
    initialIntegrationSettings={initialIntegrationSettings}>
    {children}
  </SettingsProvider>
);

export default TestProvider;
