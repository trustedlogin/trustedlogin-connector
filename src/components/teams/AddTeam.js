import { useSettings } from "../../hooks/useSettings";
import { useView } from "../../hooks/useView";
import EditTeam from "./EditTeam";
import Spinner from "../Spinner";

const AddTeam = () => {
  const { addTeam, loading } = useSettings();
  const { setCurrentView } = useView();
  const onClickSave = (newTeam) => {
    addTeam(newTeam, true);
    setCurrentView("teams");
  };
  return (
    <>
      {loading && <Spinner size={150} />}
      <EditTeam onClickSave={onClickSave} formTitle={"Add Team"} />
    </>
  );
};

export default AddTeam;
