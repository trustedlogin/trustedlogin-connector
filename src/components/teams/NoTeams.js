import { useView } from "../../hooks/useView";
import { CenteredLayout } from "../Layout";

/**
 * Displays "No Teams" message and button
 * @returns {JSX.Element}
 */
const NoTeams = () => {
  const { setCurrentView } = useView();
  return (
    <CenteredLayout
      Icon={() => (
        <>
          <svg
            className="mx-auto"
            width="56"
            height="56"
            viewBox="0 0 56 56"
            fill="none"
            xmlns="http://www.w3.org/2000/svg">
            <rect x="4" y="4" width="48" height="48" rx="24" fill="#00AADD" />
            <path
              fillRule="evenodd"
              clipRule="evenodd"
              d="M28 26C26.9391 26 25.9217 25.5786 25.1716 24.8284C24.4214 24.0783 24 23.0609 24 22C24 20.9391 24.4214 19.9217 25.1716 19.1716C25.9217 18.4214 26.9391 18 28 18C29.0609 18 30.0783 18.4214 30.8284 19.1716C31.5786 19.9217 32 20.9391 32 22C32 23.0609 31.5786 24.0783 30.8284 24.8284C30.0783 25.5786 29.0609 26 28 26ZM22 36C20.9391 36 19.9217 35.5786 19.1716 34.8284C18.4214 34.0783 18 33.0609 18 32C18 30.9391 18.4214 29.9217 19.1716 29.1716C19.9217 28.4214 20.9391 28 22 28C23.0609 28 24.0783 28.4214 24.8284 29.1716C25.5786 29.9217 26 30.9391 26 32C26 33.0609 25.5786 34.0783 24.8284 34.8284C24.0783 35.5786 23.0609 36 22 36V36ZM34 36C32.9391 36 31.9217 35.5786 31.1716 34.8284C30.4214 34.0783 30 33.0609 30 32C30 30.9391 30.4214 29.9217 31.1716 29.1716C31.9217 28.4214 32.9391 28 34 28C35.0609 28 36.0783 28.4214 36.8284 29.1716C37.5786 29.9217 38 30.9391 38 32C38 33.0609 37.5786 34.0783 36.8284 34.8284C36.0783 35.5786 35.0609 36 34 36Z"
              stroke="white"
              strokeWidth="2"
            />
            <rect
              x="4"
              y="4"
              width="48"
              height="48"
              rx="24"
              stroke="#CDEFF9"
              strokeWidth="8"
            />
          </svg>
        </>
      )}
      title={"Create your first team"}
      subtitle={"Add a support team, as configured in TrustedLogin."}>
      <button
        onClick={() => setCurrentView("teams/new")}
        type="button"
        className="inline-flex items-center justify-center mt-6 px-4 py-2 border border-transparent shadow-sm font-medium rounded-lg text-white bg-blue-tl focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500 sm:text-sm">
        <svg
          className="mr-1"
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 24 24">
          <g fill="none">
            <path
              d="M12 6v6m0 0v6m0-6h6m-6 0H6"
              stroke="#FFFFFF"
              strokeWidth="2"
              strokeLinecap="round"
              strokeLinejoin="round"></path>
          </g>
        </svg>
        Add Team
      </button>
    </CenteredLayout>
  );
};
export default NoTeams;
