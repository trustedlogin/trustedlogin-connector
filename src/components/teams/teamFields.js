import { __ } from "@wordpress/i18n";

const teamFields = {
  account_id: {
    label: __("Account ID", "trustedlogin-connector"),
    id: "account_id",
  },
  private_key: {
    label: __("Private Key", "trustedlogin-connector"),
    id: "private_key",
  },
  public_key: {
    label: __("Public Key", "trustedlogin-connector"),
    id: "public_key",
  },
  helpdesk: {
    label: __("Help Desk", "trustedlogin-connector"),
    id: "helpdesk",
    defaultValue: "helpscout",
    options: [
      { value: "helpscout", label: __("Help Scout", "trustedlogin-connector") },
      { value: "freescout", label: __("FreeScout", "trustedlogin-connector") },
    ],
  },
  approved_roles: {
    label: __("What Roles Provide Support?", "trustedlogin-connector"),
    id: "approved_roles",
    type: "array",
  },
};
export default teamFields;
