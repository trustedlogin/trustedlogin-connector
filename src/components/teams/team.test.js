import AddTeam from "./AddTeam";
import { render, act, fireEvent } from "@testing-library/react";

import NoTeams from "./NoTeams";
import EditTeam, { HelpDeskSelect } from "./EditTeam";
import TeamsList from "./TeamsList";
import TestProvider, { testTeam } from "../TestProvider";
import teamFields from "./teamFields";
import { useRef } from "react";
import collectTeam from "./collectTeam";
const team = testTeam;

const Provider = (props) => (
  <TestProvider
    {...props}
    initialIntegrationSettings={{
      helpscout: {
        enabled: true,
      },
    }}
  />
);

describe("AddTeam", () => {
  it("Renders & Matches snapshot", () => {
    const { container } = render(<AddTeam />, { wrapper: Provider });
    expect(container).toMatchSnapshot();
  });
});

describe("NoTeams", () => {
  it("Renders & Matches snapshot", () => {
    const { container } = render(<NoTeams />, { wrapper: Provider });
    expect(container).toMatchSnapshot();
  });
});

describe("EditTeam", () => {
  it("Renders & Matches snapshot", () => {
    const { container } = render(<EditTeam team={team} />, {
      wrapper: Provider,
    });
    expect(container).toMatchSnapshot();
  });
  it("Changes values and collectTeams has those value", () => {
    const onClickSave = jest.fn();
    const Test = () => {
      return <EditTeam team={team} onClickSave={onClickSave} />;
    };
    const { getByLabelText, container } = render(<Test />, {
      wrapper: Provider,
    });
    act(() => {
      fireEvent.change(getByLabelText(teamFields.account_id.label), {
        target: { value: "12345" },
      });
      fireEvent.change(getByLabelText(teamFields.private_key.label), {
        target: { value: "secret" },
      });
      fireEvent.change(getByLabelText(teamFields.public_key.label), {
        target: { value: "public" },
      });
      return;
      fireEvent.change(getByLabelText(teamFields.approved_roles.label), {
        target: { value: "administrator" },
      });
    });
    act(() => {
      fireEvent.submit(container.getElementsByTagName("form")[0]);
    });
    expect(onClickSave).toBeCalledTimes(1);
    const lastCall = onClickSave.mock.calls[0][0];
    expect(lastCall[teamFields.account_id.id]).toEqual("12345");
    expect(lastCall[teamFields.public_key.id]).toEqual("public");
    expect(lastCall[teamFields.private_key.id]).toEqual("secret");
  });
});
describe("TeamsList", () => {
  it("Renders & Matches snapshot", () => {
    const Wrapper = (props) => <Provider {...props} initialTeams={[team]} />;
    const { container } = render(<TeamsList />, {
      wrapper: Wrapper,
    });
    expect(container).toMatchSnapshot();
  });
});
describe("HelpDeskSelect", () => {
  const options = [
    { label: "Help Scout", value: "helpscout" },
    { label: "FreeScout", value: "freescout" },
  ];
  it("Renders & Matches snapshot", () => {
    const Wrapper = (props) => <Provider {...props} />;
    const { container } = render(<HelpDeskSelect />, {
      wrapper: Wrapper,
    });
    expect(container).toMatchSnapshot();
  });

  it("Has options", () => {
    const { container } = render(<HelpDeskSelect options={options} />, {
      wrapper: Provider,
    });
    expect(container.querySelectorAll("option").length)
      //Add one for placeholder
      .toBe(options.length + 1);
  });
  it("Sets default, default value", () => {
    const { getByLabelText } = render(<HelpDeskSelect options={options} />, {
      wrapper: Provider,
    });
    expect(getByLabelText(teamFields.helpdesk.label).value).toBe(
      "Select a Help Desk"
    );
  });
  it("Sets default value", () => {
    const handleChange = jest.fn();

    const { getByLabelText } = render(
      <HelpDeskSelect
        value={"helpscout"}
        options={options}
        onChange={handleChange}
      />,
      {
        wrapper: Provider,
      }
    );
    expect(getByLabelText(teamFields.helpdesk.label).value).toBe("helpscout");
  });

  it("Changes value", () => {
    const { getByLabelText } = render(<HelpDeskSelect options={options} />, {
      wrapper: Provider,
    });
    act(() => {
      fireEvent.change(getByLabelText(teamFields.helpdesk.label), {
        target: { value: "helpscout" },
      });
    });
    expect(getByLabelText(teamFields.helpdesk.label).value).toBe("helpscout");
  });

  it("Changes value and collectTeams has that value", () => {
    const fn = jest.fn();
    const Test = () => {
      const formRef = useRef();
      const onSubmit = (e) => {
        e.preventDefault();
        const team = collectTeam(formRef.current);
        fn(team);
      };

      return (
        <form ref={formRef} onSubmit={onSubmit} data-testid="test">
          <HelpDeskSelect options={options} />
        </form>
      );
    };
    const { getByLabelText, getByTestId } = render(<Test />, {
      wrapper: Provider,
    });
    act(() => {
      fireEvent.change(getByLabelText(teamFields.helpdesk.label), {
        target: { value: "freescout" },
      });
    });
    act(() => {
      fireEvent.submit(getByTestId("test"));
    });
    expect(fn).toBeCalledTimes(1);
    expect(fn).toBeCalledWith({
      [teamFields.helpdesk.id]: "freescout",
    });
  });
});
