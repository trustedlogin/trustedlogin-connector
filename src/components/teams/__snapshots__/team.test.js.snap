// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTeam Renders & Matches snapshot 1`] = `
<div>
  <form
    class="flex px-5 pt-20 sm:px-10"
  >
    <div
      class="flex flex-col w-full max-w-4xl mx-auto p-8 bg-white rounded-lg shadow sm:p-14 sm:pb-8"
    >
      <svg
        class="mx-auto"
        fill="none"
        height="56"
        viewBox="0 0 56 56"
        width="56"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#00AADD"
          height="48"
          rx="24"
          width="48"
          x="4"
          y="4"
        />
        <path
          clip-rule="evenodd"
          d="M28 26C26.9391 26 25.9217 25.5786 25.1716 24.8284C24.4214 24.0783 24 23.0609 24 22C24 20.9391 24.4214 19.9217 25.1716 19.1716C25.9217 18.4214 26.9391 18 28 18C29.0609 18 30.0783 18.4214 30.8284 19.1716C31.5786 19.9217 32 20.9391 32 22C32 23.0609 31.5786 24.0783 30.8284 24.8284C30.0783 25.5786 29.0609 26 28 26ZM22 36C20.9391 36 19.9217 35.5786 19.1716 34.8284C18.4214 34.0783 18 33.0609 18 32C18 30.9391 18.4214 29.9217 19.1716 29.1716C19.9217 28.4214 20.9391 28 22 28C23.0609 28 24.0783 28.4214 24.8284 29.1716C25.5786 29.9217 26 30.9391 26 32C26 33.0609 25.5786 34.0783 24.8284 34.8284C24.0783 35.5786 23.0609 36 22 36V36ZM34 36C32.9391 36 31.9217 35.5786 31.1716 34.8284C30.4214 34.0783 30 33.0609 30 32C30 30.9391 30.4214 29.9217 31.1716 29.1716C31.9217 28.4214 32.9391 28 34 28C35.0609 28 36.0783 28.4214 36.8284 29.1716C37.5786 29.9217 38 30.9391 38 32C38 33.0609 37.5786 34.0783 36.8284 34.8284C36.0783 35.5786 35.0609 36 34 36Z"
          fill-rule="evenodd"
          stroke="white"
          stroke-width="2"
        />
        <rect
          height="48"
          rx="24"
          stroke="#CDEFF9"
          stroke-width="8"
          width="48"
          x="4"
          y="4"
        />
      </svg>
      <div
        class="max-w-sm mx-auto mb-8 justify-center text-center"
      >
        <h2
          class="mt-4 text-2xl text-gray-900"
        >
          Add Team
        </h2>
        <p
          class="mt-2 mb-4 text-sm text-gray-500"
        />
        <a
          class="text-blue-tl text-sm"
          href="https://app.trustedlogin.com/admin/teams"
          target="_blank"
        >
          Where can I find this info?
          <span
            class="screen-reader-text"
          >
            (Link opens in a new window)
          </span>
        </a>
      </div>
      <div
        class="flex flex-col py-6 space-y-6 sm:space-y-0 sm:space-x-12 sm:flex-row"
      >
        <div
          class="flex flex-col space-y-6 sm:flex-1"
        >
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="account_id"
            >
              Account ID
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="account_id"
                name="account_id"
                required=""
                type="number"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="public_key"
            >
              Public Key
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="public_key"
                name="public_key"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="private_key"
            >
              Private Key
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="private_key"
                name="private_key"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="flex flex-col space-y-6 sm:flex-1"
        >
          <div
            class=""
          >
            <label
              class="block text-sm font-medium text-gray-700"
              for="approved_roles"
            >
              What Roles Provide Support?
            </label>
            <div
              class="mt-2"
            >
              <div
                class="multiselect-container multiSelectContainer "
                id="approved_roles"
              >
                <div
                  class="search-wrapper searchWrapper "
                >
                  <input
                    autocomplete="off"
                    class="searchBox"
                    id="approved_roles_input"
                    placeholder="Select"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="optionListContainer displayNone"
                >
                  <ul
                    class="optionContainer"
                  >
                    <li
                      class="
              highlightOption highlight 
              false 
              false option
            "
                    >
                      Administrator
                    </li>
                    <li
                      class="
               
              false 
              false option
            "
                    >
                      Editor
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <label
              class="block text-sm font-medium text-gray-700"
              for="helpdesk"
            >
              Help Desk
            </label>
            <div
              class="mt-2"
            >
              <select
                class="bg-white block w-full pl-3 pr-8 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="helpdesk"
                name="helpdesk"
              >
                <option>
                  Select a Help Desk
                </option>
                <option
                  value="helpscout"
                >
                  Help Scout
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pt-8 mt-4 border-t"
      >
        <div
          class="flex justify-end"
        >
          <button
            class="bg-white leading-6 py-2 px-4 border border-gray-300 rounded-lg text-sm font-medium text-gray-900 hover:bg-gray-50 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
            type="submit"
          >
            Cancel
          </button>
          <button
            class="ml-3 inline-flex leading-6 justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-lg text-white bg-blue-tl hover:bg-indigo-700 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
            type="submit"
          >
            Add Team
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`EditTeam Renders & Matches snapshot 1`] = `
<div>
  <form
    class="flex px-5 pt-20 sm:px-10"
  >
    <div
      class="flex flex-col w-full max-w-4xl mx-auto p-8 bg-white rounded-lg shadow sm:p-14 sm:pb-8"
    >
      <svg
        class="mx-auto"
        fill="none"
        height="56"
        viewBox="0 0 56 56"
        width="56"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#00AADD"
          height="48"
          rx="24"
          width="48"
          x="4"
          y="4"
        />
        <path
          clip-rule="evenodd"
          d="M28 26C26.9391 26 25.9217 25.5786 25.1716 24.8284C24.4214 24.0783 24 23.0609 24 22C24 20.9391 24.4214 19.9217 25.1716 19.1716C25.9217 18.4214 26.9391 18 28 18C29.0609 18 30.0783 18.4214 30.8284 19.1716C31.5786 19.9217 32 20.9391 32 22C32 23.0609 31.5786 24.0783 30.8284 24.8284C30.0783 25.5786 29.0609 26 28 26ZM22 36C20.9391 36 19.9217 35.5786 19.1716 34.8284C18.4214 34.0783 18 33.0609 18 32C18 30.9391 18.4214 29.9217 19.1716 29.1716C19.9217 28.4214 20.9391 28 22 28C23.0609 28 24.0783 28.4214 24.8284 29.1716C25.5786 29.9217 26 30.9391 26 32C26 33.0609 25.5786 34.0783 24.8284 34.8284C24.0783 35.5786 23.0609 36 22 36V36ZM34 36C32.9391 36 31.9217 35.5786 31.1716 34.8284C30.4214 34.0783 30 33.0609 30 32C30 30.9391 30.4214 29.9217 31.1716 29.1716C31.9217 28.4214 32.9391 28 34 28C35.0609 28 36.0783 28.4214 36.8284 29.1716C37.5786 29.9217 38 30.9391 38 32C38 33.0609 37.5786 34.0783 36.8284 34.8284C36.0783 35.5786 35.0609 36 34 36Z"
          fill-rule="evenodd"
          stroke="white"
          stroke-width="2"
        />
        <rect
          height="48"
          rx="24"
          stroke="#CDEFF9"
          stroke-width="8"
          width="48"
          x="4"
          y="4"
        />
      </svg>
      <div
        class="max-w-sm mx-auto mb-8 justify-center text-center"
      >
        <h2
          class="mt-4 text-2xl text-gray-900"
        >
          Update Team
        </h2>
        <p
          class="mt-2 mb-4 text-sm text-gray-500"
        />
        <a
          class="text-blue-tl text-sm"
          href="https://app.trustedlogin.com/admin/teams"
          target="_blank"
        >
          Where can I find this info?
          <span
            class="screen-reader-text"
          >
            (Link opens in a new window)
          </span>
        </a>
      </div>
      <div
        class="flex flex-col py-6 space-y-6 sm:space-y-0 sm:space-x-12 sm:flex-row"
      >
        <div
          class="flex flex-col space-y-6 sm:flex-1"
        >
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="account_id"
            >
              Account ID
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="account_id"
                name="account_id"
                required=""
                type="number"
                value="1"
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="public_key"
            >
              Public Key
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="public_key"
                name="public_key"
                required=""
                type="text"
                value="fsaffff"
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium text-gray-700"
              for="private_key"
            >
              Private Key
            </label>
            <div
              class="mt-2 relative rounded-lg"
            >
              <input
                class="block w-full pl-4 pr-10 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="private_key"
                name="private_key"
                required=""
                type="text"
                value="asda"
              />
            </div>
          </div>
        </div>
        <div
          class="flex flex-col space-y-6 sm:flex-1"
        >
          <div
            class=""
          >
            <label
              class="block text-sm font-medium text-gray-700"
              for="approved_roles"
            >
              What Roles Provide Support?
            </label>
            <div
              class="mt-2"
            >
              <div
                class="multiselect-container multiSelectContainer "
                id="approved_roles"
              >
                <div
                  class="search-wrapper searchWrapper "
                >
                  <input
                    autocomplete="off"
                    class="searchBox"
                    id="approved_roles_input"
                    placeholder="Select"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="optionListContainer displayNone"
                >
                  <ul
                    class="optionContainer"
                  >
                    <li
                      class="
              highlightOption highlight 
              false 
              false option
            "
                    >
                      Administrator
                    </li>
                    <li
                      class="
               
              false 
              false option
            "
                    >
                      Editor
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <label
              class="block text-sm font-medium text-gray-700"
              for="helpdesk"
            >
              Help Desk
            </label>
            <div
              class="mt-2"
            >
              <select
                class="bg-white block w-full pl-3 pr-8 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
                id="helpdesk"
                name="helpdesk"
              >
                <option>
                  Select a Help Desk
                </option>
                <option
                  value="helpscout"
                >
                  Help Scout
                </option>
              </select>
            </div>
          </div>
        </div>
      </div>
      <div
        class="pt-8 mt-4 border-t"
      >
        <div
          class="flex justify-end"
        >
          <button
            class="bg-white leading-6 py-2 px-4 border border-gray-300 rounded-lg text-sm font-medium text-gray-900 hover:bg-gray-50 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
            type="submit"
          >
            Cancel
          </button>
          <button
            class="ml-3 inline-flex leading-6 justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-lg text-white bg-blue-tl hover:bg-indigo-700 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
            type="submit"
          >
            Update Team
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`HelpDeskSelect Renders & Matches snapshot 1`] = `
<div>
  <div
    class=""
  >
    <label
      class="block text-sm font-medium text-gray-700"
      for="helpdesk"
    >
      Help Desk
    </label>
    <div
      class="mt-2"
    >
      <select
        class="bg-white block w-full pl-3 pr-8 py-2.5 sm:text-sm border border-gray-300 rounded-lg focus:outline-none focus:border-sky-500 focus:ring-1 ring-offset-2 focus:ring-sky-500"
        id="helpdesk"
        name="helpdesk"
      >
        <option>
          Select a Help Desk
        </option>
        <option
          value="helpscout"
        >
          Help Scout
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`NoTeams Renders & Matches snapshot 1`] = `
<div>
  <div
    class="flex px-5 pt-12 sm:px-10 sm:pt-32"
  >
    <div
      class="flex flex-col justify-center text-center w-full max-w-lg mx-auto p-8 bg-white rounded-lg shadow sm:p-14"
    >
      <svg
        class="mx-auto"
        fill="none"
        height="56"
        viewBox="0 0 56 56"
        width="56"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#00AADD"
          height="48"
          rx="24"
          width="48"
          x="4"
          y="4"
        />
        <path
          clip-rule="evenodd"
          d="M28 26C26.9391 26 25.9217 25.5786 25.1716 24.8284C24.4214 24.0783 24 23.0609 24 22C24 20.9391 24.4214 19.9217 25.1716 19.1716C25.9217 18.4214 26.9391 18 28 18C29.0609 18 30.0783 18.4214 30.8284 19.1716C31.5786 19.9217 32 20.9391 32 22C32 23.0609 31.5786 24.0783 30.8284 24.8284C30.0783 25.5786 29.0609 26 28 26ZM22 36C20.9391 36 19.9217 35.5786 19.1716 34.8284C18.4214 34.0783 18 33.0609 18 32C18 30.9391 18.4214 29.9217 19.1716 29.1716C19.9217 28.4214 20.9391 28 22 28C23.0609 28 24.0783 28.4214 24.8284 29.1716C25.5786 29.9217 26 30.9391 26 32C26 33.0609 25.5786 34.0783 24.8284 34.8284C24.0783 35.5786 23.0609 36 22 36V36ZM34 36C32.9391 36 31.9217 35.5786 31.1716 34.8284C30.4214 34.0783 30 33.0609 30 32C30 30.9391 30.4214 29.9217 31.1716 29.1716C31.9217 28.4214 32.9391 28 34 28C35.0609 28 36.0783 28.4214 36.8284 29.1716C37.5786 29.9217 38 30.9391 38 32C38 33.0609 37.5786 34.0783 36.8284 34.8284C36.0783 35.5786 35.0609 36 34 36Z"
          fill-rule="evenodd"
          stroke="white"
          stroke-width="2"
        />
        <rect
          height="48"
          rx="24"
          stroke="#CDEFF9"
          stroke-width="8"
          width="48"
          x="4"
          y="4"
        />
      </svg>
      <h2
        class="mt-4 text-2xl text-gray-900"
      >
        Create your first team
      </h2>
      <button
        class="inline-flex items-center justify-center mt-6 px-4 py-2 border border-transparent shadow-sm font-medium rounded-lg text-white bg-blue-tl focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500 sm:text-sm"
        type="button"
      >
        <svg
          class="mr-1"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
          >
            <path
              d="M12 6v6m0 0v6m0-6h6m-6 0H6"
              stroke="#FFFFFF"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </g>
        </svg>
        Add Team
      </button>
    </div>
  </div>
</div>
`;

exports[`TeamsList Renders & Matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col px-5 py-6 sm:px-10"
  >
    <div
      class="pb-6 mb-6 border-b md:flex md:items-center md:justify-between md:space-x-5"
    >
      <div
        class="flex items-center space-x-5"
      >
        <div>
          <h2
            class="text-2xl text-gray-900"
          >
            Teams
          </h2>
          <p
            class="mt-1 text-sm text-gray-500"
          >
            Manage your TrustedLogin settings
          </p>
        </div>
      </div>
      <div
        class="mt-6 flex flex-col-reverse justify-stretch space-y-4 space-y-reverse sm:flex-row-reverse sm:justify-end sm:space-x-reverse sm:space-y-0 sm:space-x-3 md:mt-0 md:flex-row md:space-x-3"
      >
        <button
          class="ml-3 inline-flex leading-6 justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-lg text-white bg-blue-tl hover:bg-indigo-700 focus:outline-none focus:ring-2 ring-offset-2 focus:ring-sky-500"
          type="submit"
        >
          <svg
            class="mr-1"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
            >
              <path
                d="M12 6v6m0 0v6m0-6h6m-6 0H6"
                stroke="#FFFFFF"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </g>
          </svg>
          Add Team
        </button>
      </div>
    </div>
    <div
      class="flex flex-col justify-center w-full bg-white rounded-lg shadow"
    >
      <ul
        class="divide-y divide-gray-200 px-5 sm:px-8"
        role="list"
      >
        <li
          class="py-5 flex flex-col items-center justify-between sm:py-8 sm:flex-row"
        >
          <div
            class="flex w-full items-center space-x-5 sm:w-auto"
          >
            <button
              class="flex-shrink-0 flex items-center justify-center h-12 w-12 bg-purple-600 text-white text-sm font-medium rounded-lg"
            >
              TL
            </button>
            <div
              class="flex flex-row space-x-16 items-center w-full justify-between sm:justify-start"
            >
              <div
                class="flex flex-col sm:flex-row sm:items-center max-w-[10rem] sm:max-w-[8rem] md:max-w-none"
              >
                <div
                  class="flex flex-col"
                >
                  <p
                    class="text-lg font-medium text-gray-900 leading-tight min-w-[6rem]"
                    id="team-option-1-label"
                  >
                    Team 0
                  </p>
                  <p
                    class="text-sm text-gray-500"
                    id="team-option-1-description"
                  >
                    #
                    1
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex items-center space-x-5 w-full mt-4 justify-between sm:w-auto sm:mt-0"
          >
            <div
              class="flex items-center space-x-6"
            >
              <button
                class="text-sm text-blue-tl hover:text-navy-tl p-2"
              >
                Edit
              </button>
              <button
                class="text-sm text-blue-tl hover:text-navy-tl p-2"
              >
                <svg
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    stroke="none"
                  />
                  <path
                    d="M20 11a8.1 8.1 0 0 0 -15.5 -2m-.5 -5v5h5"
                  />
                  <path
                    d="M4 13a8.1 8.1 0 0 0 15.5 2m.5 5v-5h-5"
                  />
                </svg>
              </button>
              <button
                class="text-sm text-red-500 hover:text-red-800 p-2"
              >
                Delete
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
